### DISCUSSION TITLE
[@types/testing-library__jest-dom] hijacks Jest's `Matchers` interface

### DISCUSSION THREAD

**Discussion by vhenzl** [DefinitelyTyped/DefinitelyTyped#64858] [Issues with a @types package]
The `Matchers` interface in `@types/testing-library__jest-dom` currently extends `Record<string, any>` interface (via `TestingLibraryMatchers` interface): https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/testing-library__jest-dom/matchers.d.ts#L1

This makes the `Matchers` interface unsafe. Because of `Record`, any unknown property or function is automatically `any`.

So, if I make a typo like `expect('a value').toBee()` (instead of `toBe()`) or use a non-existent matcher `expect('a value').toBeXYZ()`, TypeScript can't point it out, and I learn about the problem only in runtime when running the test.

There seems not to be any good reason for extending from `Record`; it should be removed from the type.

[a comment by an automated account is not shown]

**Comment by kryops** [comment]
We also noticed this, and created a PR with a fix: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66247

Just removing the `Record<string, any>` wasn't enough as it would break the usage of `expect.extend(matchers)`.
However, the type of `matchers` can be easily separated from the global interface that actually extends the type of `expect`.

**Reply by vhenzl** [reply]
Great, thanks for that!
